Web Inspector: (Regression: 254636@main) Mini console always opens when choosing "Inspect Element", even if it was previously closed

patrickangle opened the pull request:
#### ed99b93877dd9c47942db5a0f56a82acc4adeebd
<pre>
Web Inspector: (Regression: 254636@main) Mini console always opens when choosing &quot;Inspect Element&quot;, even if it was previously closed
<a href="https://bugs.webkit.org/show_bug.cgi?id=253273">https://bugs.webkit.org/show_bug.cgi?id=253273</a>
rdar://106260652

Reviewed by Devin Rousso.

When console snippets were introduced, the function signature of `appendImmediateExecutionWithResult` was updated so
that optional values were part of an options object. However, in changing that it appears it was assumed that the
default would be flipped to `false`, but that never happened. The parameter was no longer being provided by DOMManger
when inspecting an element, and therefor we fell back to the default of showing the quick console.

To fix this, we correct the `shouldRevealConsole` to be `false` by default, matching other option objects, and then
clean up other places where we were assuming it would still be true (mostly anywhere that a &quot;Log [thing]&quot; option was
present).

* Source/WebInspectorUI/UserInterface/Controllers/JavaScriptLogViewController.js:
(WI.JavaScriptLogViewController.prototype.appendImmediateExecutionWithResult.saveResultCallback):
(WI.JavaScriptLogViewController.prototype.appendImmediateExecutionWithResult):
- Use the new options object of the ConsoleCommandResultMessage constructor.

* Source/WebInspectorUI/UserInterface/Models/ConsoleCommandResultMessage.js:
- Make `shouldRevealConsole` part of an options object, and make it default to false instead.

* Source/WebInspectorUI/UserInterface/Views/AnimationContentView.js:
(WI.AnimationContentView.prototype._populateAnimationTargetButtonContextMenu):
(WI.AnimationContentView):
* Source/WebInspectorUI/UserInterface/Views/CanvasContentView.js:
(WI.CanvasContentView.prototype._populateCanvasElementButtonContextMenu):
* Source/WebInspectorUI/UserInterface/Views/CanvasTreeElement.js:
(WI.CanvasTreeElement.prototype.populateContextMenu):
* Source/WebInspectorUI/UserInterface/Views/ContextMenuUtilities.js:
* Source/WebInspectorUI/UserInterface/Views/ObjectPreviewView.js:
(WI.ObjectPreviewView.prototype._contextMenuHandler):
(WI.ObjectPreviewView):
* Source/WebInspectorUI/UserInterface/Views/ObjectTreeBaseTreeElement.js:
(WI.ObjectTreeBaseTreeElement.prototype._logSymbolProperty):
(WI.ObjectTreeBaseTreeElement.prototype._logValue):
* Source/WebInspectorUI/UserInterface/Views/QuickConsole.js:
(WI.QuickConsole.prototype._handleDrop):
* Source/WebInspectorUI/UserInterface/Views/WebSocketResourceTreeElement.js:
(WI.WebSocketResourceTreeElement.prototype.populateContextMenu):

Canonical link: <a href="https://commits.webkit.org/262106@main">https://commits.webkit.org/262106@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c527a774b4689d6b0032b86a96e08275195c7d18

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/572 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/590 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/616 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/686 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/497 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/638 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/675 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/740 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/580 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/550 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/547 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/685 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/615 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/534 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/574 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/506 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/565 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/540 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/492 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/549 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/136 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/549 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Controllers/JavaScriptLogViewController.js:
```diff
@@ -138,7 +138,7 @@ WI.JavaScriptLogViewController = class JavaScriptLogViewController extends WI.Ob
         function saveResultCallback(savedResultIndex)
         {
-            let commandResultMessage = new WI.ConsoleCommandResultMessage(result.target, result, false, savedResultIndex, shouldRevealConsole);
+            let commandResultMessage = new WI.ConsoleCommandResultMessage(result.target, result, false, savedResultIndex, !!shouldRevealConsole);
```
i think it'd be better to have the `!!` be inside `WI.ConsoleCommandResultMessage` instead to catch all cases it could be provided instead of just this one

dcrousso commented on Source/WebInspectorUI/UserInterface/Models/ConsoleCommandResultMessage.js:
```diff
@@ -29,7 +29,7 @@
 
 WI.ConsoleCommandResultMessage = class ConsoleCommandResult extends WI.ConsoleMessage
 {
-    constructor(target, result, wasThrown, savedResultIndex, shouldRevealConsole = true)
```
a perfect example of why i dont like parameter defaults (other than `= { }`) 🥲

perhaps we should turn this into an `{ shouldRevealConsole } = { }` for `WI.ConsoleCommandResultMessage ` as well?

dcrousso reviewed (approved):
i wonder if we even should be logging to the Console in the first place when you Inspect Element.  that's kinda why we have the Log Element contextmenu item (and `$0`) 🤔

typo in commit message
> and therefor we

patrickangle committed:
Web Inspector: (Regression: 254636@main) Mini console always opens when choosing "Inspect Element", even if it was previously closed
https://bugs.webkit.org/show_bug.cgi?id=253273
rdar://106260652

Reviewed by Devin Rousso.

When console snippets were introduced, the function signature of `appendImmediateExecutionWithResult` was updated so
that optional values were part of an options object. However, in changing that it appears it was assumed that the
default would be flipped to `false`, but that never happened. The parameter was no longer being provided by DOMManger
when inspecting an element, and therefor we fell back to the default of showing the quick console.

To fix this, we correct the `shouldRevealConsole` to be `false` by default, matching other option objects, and then
clean up other places where we were assuming it would still be true (mostly anywhere that a "Log [thing]" option was
present).

* Source/WebInspectorUI/UserInterface/Controllers/JavaScriptLogViewController.js:
(WI.JavaScriptLogViewController.prototype.appendImmediateExecutionWithResult.saveResultCallback):
(WI.JavaScriptLogViewController.prototype.appendImmediateExecutionWithResult):
- Use the new options object of the ConsoleCommandResultMessage constructor.

* Source/WebInspectorUI/UserInterface/Models/ConsoleCommandResultMessage.js:
- Make `shouldRevealConsole` part of an options object, and make it default to false instead.

* Source/WebInspectorUI/UserInterface/Views/AnimationContentView.js:
(WI.AnimationContentView.prototype._populateAnimationTargetButtonContextMenu):
(WI.AnimationContentView):
* Source/WebInspectorUI/UserInterface/Views/CanvasContentView.js:
(WI.CanvasContentView.prototype._populateCanvasElementButtonContextMenu):
* Source/WebInspectorUI/UserInterface/Views/CanvasTreeElement.js:
(WI.CanvasTreeElement.prototype.populateContextMenu):
* Source/WebInspectorUI/UserInterface/Views/ContextMenuUtilities.js:
* Source/WebInspectorUI/UserInterface/Views/ObjectPreviewView.js:
(WI.ObjectPreviewView.prototype._contextMenuHandler):
(WI.ObjectPreviewView):
* Source/WebInspectorUI/UserInterface/Views/ObjectTreeBaseTreeElement.js:
(WI.ObjectTreeBaseTreeElement.prototype._logSymbolProperty):
(WI.ObjectTreeBaseTreeElement.prototype._logValue):
* Source/WebInspectorUI/UserInterface/Views/QuickConsole.js:
(WI.QuickConsole.prototype._handleDrop):
* Source/WebInspectorUI/UserInterface/Views/WebSocketResourceTreeElement.js:
(WI.WebSocketResourceTreeElement.prototype.populateContextMenu):

Canonical link: https://commits.webkit.org/262106@main
